### ISSUE TITLE
What exactly happens when one fixes the seed before importing keras? The variations, if seed is not fixed, can be used to estimate uncetainties of the model?

### ISSUE THREAD

**Issue by mmelodea** [keras-team/keras#9808]
Hi guys,

I've 2 conceptual questions about keras. Hope is not a problem to ask them in the same time (since they are related). 

First: In many examples and also there are some questions about keras reproducibility of results, appears the motivation to fix a seed before one imports anything related to keras. I know keras randomly sets the events for each batch during the fit and that if not fixed could lead to different results in the end of training even using the same configuration in different runs. So, could some one explain what exactly happens when we fix the seed? Is there any other stochastic process that is affected by fixing the seed? I looked in keras documentation and the questions already asked here but I didn't find a clear detailing for that.

Second: These variations we observed in the final models, when not fixing the seed, can be used as an estimation of the uncertainty of the keras (for a particular application)?

Thanks in advance!
And I'm sorry if I should have asked the questions separately.

**Comment by gabrieldemarmiesse** [comment]
First question: Yes, keras uses the numpy seed to do quite some things, I can't list them all, but shuffling the  dataset in `fit` is one of them. There can be many more.
BUT! The numpy seed doesn't affect the seed sometimes used by the backend (eg theano or tensorflow) because they might use a different seed internally. So fixing the seed in numpy doesn't garantee perfect reproducibility, you have to go deeper into how your backend generate random numbers.

second question: No, don't use this. There would be nothing that would garantee the results. There is a nice blog post about computing the confidence of a network though: 

http://mlg.eng.cam.ac.uk/yarin/blog_3d801aa532c1ce.html


hope it helps.

**Comment by mmelodea** [comment]
Hi gabrieldemarlesse,

Thank you for reply!
Yes, your answer does help. That blog post is very good!
I'm closing the issue.
